lib/apiservers/engine/backends/container.go

```diff
@@ -20,88 +20,113 @@ import (
 	"io"
 	"net/http"
 	"os"
-	goexec "os/exec"
+	"os/exec"
 	"sync"
 	"time"
 
+	"golang.org/x/net/context"
+
+	"github.com/go-swagger/go-swagger/httpkit"
+	httptransport "github.com/go-swagger/go-swagger/httpkit/client"
+
 	log "github.com/Sirupsen/logrus"
 	"github.com/docker/docker/api/types/backend"
 	derr "github.com/docker/docker/errors"
 	"github.com/docker/docker/pkg/archive"
+	"github.com/docker/docker/pkg/stdcopy"
 	"github.com/docker/docker/pkg/version"
 	"github.com/docker/engine-api/types"
 	"github.com/docker/engine-api/types/container"
 	"github.com/docker/engine-api/types/strslice"
 
+	viccontainer "github.com/vmware/vic/lib/apiservers/engine/backends/container"
+	"github.com/vmware/vic/lib/apiservers/portlayer/client"
 	"github.com/vmware/vic/lib/apiservers/portlayer/client/containers"
+	"github.com/vmware/vic/lib/apiservers/portlayer/client/interaction"
 	"github.com/vmware/vic/lib/apiservers/portlayer/client/scopes"
 	"github.com/vmware/vic/lib/apiservers/portlayer/client/storage"
 	"github.com/vmware/vic/lib/apiservers/portlayer/models"
 	"github.com/vmware/vic/lib/guest"
 	"github.com/vmware/vic/pkg/trace"
 )
 
-type V1Compatibility struct {
-	ID     string
-	Config container.Config
-}
-
 type Container struct {
 	ProductName string
-
-	// Protects following map
-	m sync.Mutex
-
-	// FIXME: in-memory map to keep image name to vmdk name relationship
-	HackMap map[string]V1Compatibility
 }
 
 // docker's container.execBackend
 
+// ContainerExecCreate sets up an exec in a running container.
 func (c *Container) ContainerExecCreate(config *types.ExecConfig) (string, error) {
 	return "", fmt.Errorf("%s does not implement container.ContainerExecCreate", c.ProductName)
 }
 
+// ContainerExecInspect returns low-level information about the exec
+// command. An error is returned if the exec cannot be found.
 func (c *Container) ContainerExecInspect(id string) (*backend.ExecInspect, error) {
 	return nil, fmt.Errorf("%s does not implement container.ContainerExecInspect", c.ProductName)
 }
 
+// ContainerExecResize changes the size of the TTY of the process
+// running in the exec with the given name to the given height and
+// width.
 func (c *Container) ContainerExecResize(name string, height, width int) error {
-	return fmt.Errorf("%s does not implement container.ContainerExecResize", c.ProductName)
+	return nil
+	//	return fmt.Errorf("%s does not implement container.ContainerExecResize", c.ProductName)
 }
 
+// ContainerExecStart starts a previously set up exec instance. The
+// std streams are set up.
 func (c *Container) ContainerExecStart(name string, stdin io.ReadCloser, stdout io.Writer, stderr io.Writer) error {
 	return fmt.Errorf("%s does not implement container.ContainerExecStart", c.ProductName)
 }
 
+// ExecExists looks up the exec instance and returns a bool if it exists or not.
+// It will also return the error produced by `getConfig`
 func (c *Container) ExecExists(name string) (bool, error) {
```

Comment:
Yes, name encompasses Id, friendly name, partial id, etc.  These are the same signature that Docker uses for their backend interfaces.  We can make a pass and clean up the method signature to make it clear, but I'd like to do that in another PR, outside of this one.